extensions/typescript-language-features/src/languageFeatures/tsconfig.ts

```diff
@@ -110,6 +114,55 @@ class TsconfigLinkProvider implements vscode.DocumentLinkProvider {
 	}
 }
 
+// Reference: https://github.com/microsoft/TypeScript/blob/febfd442cdba343771f478cf433b0892f213ad2f/src/compiler/commandLineParser.ts#L3005
+/**
+ * @returns Returns undefined in case of lack of result while trying to resolve from node_modules
+ */
+const getTsconfigPath = async (baseDirUri: vscode.Uri, extendsValue: string): Promise<vscode.Uri | undefined> => {
+	// Don't take into account a case, where tsconfig might be resolved from the root (see the reference)
+	// e.g. C:/projects/shared-tsconfig/tsconfig.json (note that C: prefix is optional)
+
+	const isRelativePath = ['./', '../'].some(str => extendsValue.startsWith(str));
+	if (isRelativePath) {
+		const absolutePath = vscode.Uri.joinPath(baseDirUri, extendsValue);
+		if (await exists(absolutePath)) { return absolutePath; }
+		// Will suggest to create a .json variant if it doesn't exist yet
+		return absolutePath.with({
+			path: `${absolutePath.path}.json`
+		});
+	}
+
+	// Otherwise resolve like a module
+	return resolveNodeModulesPath([
+		extendsValue,
+		`${extendsValue}.json`,
+		`${extendsValue}/tsconfig.json`,
+	]);
+
+	async function resolveNodeModulesPath(pathCandidates: string[]): Promise<vscode.Uri | undefined> {
+		let currentUri = baseDirUri;
+		while (true) {
+			const nodeModulesUri = vscode.Uri.joinPath(currentUri, 'node_modules');
+			let nodeModulesStat: vscode.FileStat | undefined;
+			try {
+				nodeModulesStat = await vscode.workspace.fs.stat(nodeModulesUri);
+			} catch (err) { }
+			if (nodeModulesStat && (nodeModulesStat.type & vscode.FileType.Directory)) {
+				for (const uriCandidate of pathCandidates.map((relativePath) => vscode.Uri.joinPath(nodeModulesUri, relativePath))) {
+					if (await exists(uriCandidate)) {
+						return uriCandidate;
+					}
+				}
+			}
+			// reached the root
+			if (posix.relative(currentUri.path, '/') === '') { return; }
```

Comment:
This should stop when we reach the root of the workspace, not the root of the user's file system

Reply:
Is there any reason for doing this? I don't use `getWorkspaceFolder` and `posix.relative(currentUri.path, currentWorkspacePath.uri.path) === ''` on purpose as this should conform Node resolving algorithm, it was a purpose of this PR!

For example what if someone wants to open `packages/some-package` with structure from https://github.com/microsoft/vscode/pull/141062#issuecomment-1017432148? tssever won't stop at root of the workspace, why we should??

Also, you have already started discussion about that that above.

Reply:
Also, WebStorm works this way..

Reply:
I fixed a bug with inifite loop, so now error message displayed correctly. The reason was that `path.relative('C:/', '/')` never gives an empty string.